tango_simlib/tests/test_fandango_json_parser.py

```diff
@@ -0,0 +1,196 @@
+    'data_type': tango._tango.CmdArgType.DevString,
+    'database': 'monctl:10000',
+    'description': '',
+    'device': 'sys/database/2',
+    'display_unit': 'No display unit',
+    'events': {
+        'arch_event': {
+            'archive_abs_change': 'Not specified',
+            'archive_period': 'Not specified',
+            'archive_rel_change': 'Not specified',
+            'extensions': '[]'
+            },
+        'ch_event': {
+            'abs_change': 'Not specified',
+            'extensions': '[]',
+            'rel_change': 'Not specified'
+            },
+        'per_event': {
+            'extensions': '[]',
+            'period': '1000'
+            }
+        },
+    'format': '%s',
+    'label': 'Status',
+    'max_alarm': 'Not specified',
+    'min_alarm': 'Not specified',
+    'model': 'monctl:10000/sys/database/2/Status',
+    'name': 'Status',
+    'polling': 0,
+    'quality': 'ATTR_VALID',
+    'standard_unit': 'No standard unit',
+    'string': 'Device is OK',
+    'time': 1521724934.799581,
+    'unit': '',
+    'value': 'Device is OK',
+    'writable': 'READ'
+    }
+
+
+class GenericSetup(unittest.TestCase):
+    """A class providing the setUp method definition for the other test classes.
+    """
+    longMessage = True
+
+    def setUp(self):
+        super(GenericSetup, self).setUp()
+        self.fandango_json_file = pkg_resources.resource_filename(
+            'tango_simlib.tests.config_files', 'database2.json')
+        self.fandango_parser = fandango_json_parser.FandangoExportDeviceParser()
+        self.fandango_parser.parse(self.fandango_json_file)
+
+
+class TestFandangoJsonParser(GenericSetup):
+    """A test class that tests that the fandango json parser works correctly.
+    """
+    def test_preprocess_command_types(self):
+        """Testing that command property mapping was done well and command type
+        of mapped properties is of tango type.
+        """
+
+        # testing that the command properties have been renamed to
+        # match values in command signature
+        for cmd_name in self.fandango_parser.get_device_command_metadata():
+            self.assertEquals(
+                EXPECTED_CMD_PROPERTY_PARAMETERS.sort(),
+                self.fandango_parser.get_device_command_metadata()
+                [cmd_name].keys().sort(),
+                "The mapping of the command properties to the command"
+                "signature wasn't successful for '%s' command" % (cmd_name)
+                )
+            # testing that data type of cmd property values is a tango type and
+            # there is no occurence of 'Const' in the value of the cmd property
+            for cmd_prop_value in [
+                    self.fandango_parser.get_device_command_metadata()
+                    [cmd_name][INTERESTED_CMD_PROP[0]],
+                    self.fandango_parser.get_device_command_metadata()
+                    [cmd_name][INTERESTED_CMD_PROP[1]]]:
```

Comment:
Good that these long lines were were split, but the point at which the lines were split makes it a little difficult to see that the lines carry on.  Rather leave a trailing bracket on the previous line, and then indent below.  There are a few options, here is one (still not pretty):
```
for cmd_prop_value in [
        self.fandango_parser.get_device_command_metadata()[cmd_name][
            INTERESTED_CMD_PROP[0]],
        self.fandango_parser.get_device_command_metadata()[cmd_name][
            INTERESTED_CMD_PROP[1]]]:
    self.assertEquals( ...
```

Using @kmadia's earlier suggestion of iterating through the `items()` instead, would also help make the lines much shorter:
```
for cmd_prop_value in [
        cmd_metadata[INTERESTED_CMD_PROP[0]],
        cmd_metadata[INTERESTED_CMD_PROP[1]]]:
    self.assertEquals(...
```
Having a multiline list as part of the for loop still looks a bit messy to me (just personal taste), so I'd rather define the list beforehand:
```
cmd_prop_values = [
    cmd_metadata[INTERESTED_CMD_PROP[0]],
    cmd_metadata[INTERESTED_CMD_PROP[1]]]
for cmd_prop_value in cmd_prop_values:
    self.assertEquals(...
```